packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1725,8 +1725,33 @@ class Transform extends SingleChildRenderObjectWidget {
   /// The origin of the coordinate system (relative to the upper left corner of
   /// this render object) in which to apply the matrix.
   ///
+  /// IMPORTANT: This offset is applied AFTER any [alignment] transformation.
+  /// Since [alignment] defaults to [Alignment.center], the origin offset will be
+  /// relative to the center of the widget, not the top-left corner, unless
+  /// alignment is explicitly modified.
+  ///
+  /// Example of unexpected behavior with default alignment:
+  /// ```dart
+  /// Transform.rotate(
+  ///   angle: math.pi,
+  ///   origin: const Offset(100, 100), // Offset from center, not top-left
+  ///   child: const Text("I am rotating off of the center"),
+  /// )
+  /// ```
+  ///
+  /// To make origin truly relative to top-left corner:
```

Comment:
"In this example, the origin does appear to be offset relative to the top-left corner."